lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -0,0 +1,117 @@
+const { errors } = require('arsenal');
+/**
+ * Calculates retain until date for the locked object version
+ * @param {object} retention - includes days or years retention period
+ * @return {object} the date until the object version remains locked
+ */
+function calculateRetainUntilDate(retention) {
+    const { days, years } = retention;
+    const date = new Date();
+    // Calculate the number of days to retain the lock on the object
+    const retainUntilDays = days || years * 365;
+    const retainUntilDate
+        = date.setDate(date.getDate() + retainUntilDays);
+    return new Date(retainUntilDate);
+}
+
+/**
+ * Validates object lock headers
+ * @param {object} bucket - bucket metadata
+ * @param {object} headers - request headers
+ * @param {object} log - the log request
+ * @return {object} - object with error if validation fails
+ */
+function validateHeaders(bucket, headers, log) {
```

Comment:
Same as the other PR, this doesn't need to return `{ error: {...} }` it could just return `{...}` of the error and null otherwise.

Reply:
fixed with a rebase